lib/bpl/transformation/shadowing.rb

```diff
@@ -26,6 +26,8 @@ def shadow_copy(node)
         shadow
       end
 
+      def shadow_expr_eq(expr) "#{expr} == #{shadow_copy(expr)}" end
```

Comment:
Ah, haha, good point!

Two options then:

1. convert those strings to Nodes first via the `bpl` method — fixing the argument type of `shadow_eq` to be `Node`.

2. allow the argument type to be `Node` or `string`, and handle both cases in `shadow_copy`.